astroid/inference.py

```diff
@@ -80,14 +80,16 @@ def infer_end(
 
 # We add ignores to all assignments to methods
 # See https://github.com/python/mypy/issues/2427
-nodes.Module._infer = infer_end  # type: ignore[assignment]
-nodes.ClassDef._infer = infer_end  # type: ignore[assignment]
+nodes.Module._infer = infer_end
+nodes.ClassDef._infer = infer_end
 nodes.Lambda._infer = infer_end  # type: ignore[assignment]
 nodes.Const._infer = infer_end  # type: ignore[assignment]
 nodes.Slice._infer = infer_end  # type: ignore[assignment]
 
 
-def _infer_sequence_helper(node, context: InferenceContext | None = None):
+def _infer_sequence_helper(
+    node: _BaseContainerT, context: InferenceContext | None = None
+) -> list[nodes.NodeNG]:
```

Comment:
Is `node.elts` correctly typed?

Reply:
`BaseContainer` has `self.elts: list[NodeNG] = []`

Reply:
Hmm that sounds weird. I need to investigate this but I'm fairly certain that it could be `Instance` as well.

Reply:
I checked and this is likely incorrect. In the construction of `Lists` in `DictModel.attr_values` we use the `keys` of a `Dict.items`. Those are `SuccessfulInferenceResults`.
So `elts` is indeed incorrectly typed.

Reply:
Should this change be undone?

Reply:
Can't you change it to `SuccessfulInferenceResult`?